### PR TITLE
CategoryTree, Categories and Paths

### PR DESCRIPTION
**Pull request by dgon-jd**

### CODE REVIEW

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs [hunk 1]
```diff
@@ -159,6 +169,171 @@ public async Task TestCreateInstanceAndQuery()
...
+            return;
+        }
+        var (createdTree, createdCategory1, createdCategory12, createdCategory13, createdCategory121, createdCategory1211) = await BuildTestTreeAsync();
+
+        var list = await _eavService.GetCategoryTreeViewAsync(createdTree.Id, CancellationToken.None);
+        
+        list.Should().Contain(x => x.Id == createdCategory1.Id);
+        var instance1 = list.FirstOrDefault(x => x.Id == createdCategory1.Id);
+        instance1.Children.Should().Contain(x => x.Id == createdCategory13.Id);
+        var instance12 = instance1.Children.FirstOrDefault(x => x.Id == createdCategory12.Id);
```
Comment (ustims) [inline CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs:248]: Please give those instances better names, it's really hard to follow. Something from real world, ecommerce examples will work, like Computers->Laptops|Desktops->PC|Mac

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs [hunk 2]
```diff
@@ -22,14 +22,24 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+// ReSharper disable AsyncConverter.ConfigureAwaitHighlighting
 
 namespace CloudFabric.EAV.Tests;
 
+public enum ProjectionStorageType
```
Comment (ustims) [inline CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs:29]: There is no need to use such enum. There is a test class for every storage implementation, you can simply override test method there and add [Ignore] attribute

File: CloudFabric.EAV.Tests/Tests.cs [hunk 3]
```diff
@@ -1,2365 +1,2373 @@
-using System.Globalization;
```
Comment (ustims) [inline CloudFabric.EAV.Tests/Tests.cs:1]: Could you please make your changes in such a way that does not overwrite whole file history?

**Review by ustims — CHANGES_REQUESTED** [review]

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs [hunk 4]
```diff
@@ -22,14 +22,24 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+// ReSharper disable AsyncConverter.ConfigureAwaitHighlighting
```
Comment (ustims) [inline CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs:25]: Please move category tests out to a new file

**Review by ustims — CHANGES_REQUESTED** [review]

File: CloudFabric.EAV.Service/EAVService.cs [hunk 5]
```diff
@@ -790,25 +989,48 @@ public async Task<(EntityInstanceViewModel, ProblemDetails)> UpdateEntityInstanc
...
-        var results = await projectionRepository.Query(query, entityConfigurationId.ToString(), cancellationToken);
+        var results = await projectionRepository.Query(query, entityConfigurationId.ToString(), cancellationToken)
+            .ConfigureAwait(false);
 
         return results.TransformResultDocuments(
             r => _entityInstanceFromDictionaryDeserializer.Deserialize(entityConfiguration, attributes, r)
         );
     }
 
+    public async Task<(EntityInstanceViewModel, ProblemDetails)> MoveItemAsync(Guid itemId, string itemPartitionKey, Guid treeId, Guid newParentId, CancellationToken cancellationToken)
```
Comment (ustims) [inline CloudFabric.EAV.Service/EAVService.cs:1012]: `Move` is too wide and does not include anything related to categories. And for PIM, moving can mean a lot - moving between warehouses for example. Let's just say UpdateCategoryPath.

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs (hunk 1 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs]: Changed to the laptops example

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs (hunk 2 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs]: Done

File: CloudFabric.EAV.Service/EAVService.cs (hunk 5 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Service/EAVService.cs]: Renamed

File: CloudFabric.EAV.Tests/Tests.cs (hunk 3 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Tests/Tests.cs]: Reverted to LF line endings

File: CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs (hunk 4 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Tests/EntityInstanceQueryingTests.cs:12]: Moved to a separate test file

**Review by ustims — APPROVED** [review]
